ado_asana_sync/sync/sync.py

```diff
@@ -125,44 +241,72 @@ def sync_project(a: app, project):
         # Skip this item if is not assigned, or the assignee does not match an Asana user.
         ado_assigned_email = get_task_user_email(ado_task)
         if ado_assigned_email == None:
-            logging.info(
-                f"skipping item as it is not assigned {ado_task.fields['System.Title']}"
+            logging.debug(
+                f"{ado_task.fields['System.Title']}:skipping item as it is not assigned"
             )
             continue
         asana_matched_user = matching_user(asana_users, ado_assigned_email)
         if asana_matched_user == None:
             continue
 
-        current_work_item = work_item(
-            ado_id=ado_task.id,
-            title=ado_task.fields["System.Title"],
-            description=ado_task.fields.get("System.Description"),
-            status=ado_task.fields["System.State"],
-            item_type=ado_task.fields["System.WorkItemType"],
-            created_date=ado_task.fields["System.CreatedDate"],
-            priority=ado_task.fields["Microsoft.VSTS.Common.Priority"],
-            url=ado_task.url,
-            assigned_to=asana_matched_user.gid,
-        )
-
-        # Get the corresponding Asana task by name
-        asana_task = get_asana_task(a, asana_project, current_work_item.asana_title)
-        if asana_task == None:
-            # The Asana task does not exist, create it
-            logging.info(f"creating task {current_work_item.asana_title}")
-            create_asana_task(
-                a,
-                asana_project,
-                current_work_item,
-            )
-        else:
-            # The Asana task exists, update it
-            logging.info(f"updating task {current_work_item.asana_title}")
-            update_asana_task(
-                a,
-                asana_task.gid,
-                current_work_item,
+        # Check if this is an already mapped item.
+        item = TaskItem.find_by_ado_id(a, ado_task.id)
+        if item == None:
+            logging.info(f"{ado_task.fields['System.Title']}:unmapped task")
+            item = TaskItem(
+                ado_id=ado_task.id,
+                ado_rev=ado_task.rev,
+                title=ado_task.fields["System.Title"],
+                item_type=ado_task.fields["System.WorkItemType"],
+                created_date=iso8601_utc(datetime.now()),
+                updated_date=iso8601_utc(datetime.now()),
```

Comment:
**Suggestion:** Consider using the `datetime.utcnow()` method instead of `datetime.now()` to get the current date and time. This will return the current UTC date and time, which is more suitable for use in timestamping and other time-related operations in a distributed system.
```suggestion
                item.created_date = iso8601_utc(datetime.utcnow()),
                item.updated_date = iso8601_utc(datetime.utcnow()),
```